### ISSUE TITLE
Invalid timezone specifier in date format in rest/build.go

### ISSUE THREAD

**Issue by shabbyrobe** [aws/aws-sdk-go-v2#178]
### Version of AWS SDK for Go?
v2.0.0-preview.3

This is a copy-paste of the same issue from v1. Not sure whether this repo has a separate team, or what the internal process is for managing bugs between v1 and v2, so I'm reporting in both places to make sure. https://github.com/aws/aws-sdk-go/issues/1937

### Version of Go (`go version`)?
1.10

### What issue did you see?
Line 24 of [rest/build.go](https://github.com/aws/aws-sdk-go-v2/blob/ac26486d921d336e20fec49e2afb348c801f0003/private/protocol/rest/build.go#L24) shows the following date format string:

```
const RFC822 = "Mon, 2 Jan 2006 15:04:05 GMT"
```

The same problematic format also appears in [private/protocol/fields.go](https://github.com/aws/aws-sdk-go-v2/blob/39ee1516bc1a9781307bd0249a3e0ce5946f9fbf/private/protocol/fields.go#L138) in v2.

`GMT` is not a valid timezone specifier in a date format string, it should be `MST`.

### Steps to reproduce

```go
package main

import (
	"fmt"
	"time"
)

const RFC822 = "Mon, 2 Jan 2006 15:04:05 GMT"

func main() {
	// Works:
	fmt.Println(time.Parse("Mon, 2 Jan 2006 15:04:05 MST", "Fri, 18 May 2018 01:30:31 AEST"))
	
	// Doesn't work:
	fmt.Println(time.Parse("Mon, 2 Jan 2006 15:04:05 GMT", "Fri, 18 May 2018 01:30:31 AEST"))
}
```

**Comment by jasdel** [comment]
Tracking this discussion in the v1 SDK's aws/aws-sdk-go#1937 issue.

**Comment by skmcgrail** [comment]
See https://github.com/aws/aws-sdk-go/pull/1939 for fix.

**Comment by shabbyrobe** [comment]
This should not be closed if it has not been fixed. It was reported as fixed in v1 in aws/aws-sdk-go#1939, but those changes do not appear to have made it into v2.

**Comment by jasdel** [comment]
The v2 SDK v0.25.0 update significantly refactored the SDK's (de)serialization of API parameters. This refactor changed the SDK's deserialization behavior with regard to time values to adhere to the SDK's API protocol definitions. The SDK will always serialize UTC time, and will expect the response time values to adhere to the API protocol's defined format. In the case of the RFC1123 format, `GMT` timezone is the value expected.
